buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java

```diff
@@ -120,23 +120,25 @@ public String call() throws Exception {
                 });
             });
         });
-
         publishing.getPublications().withType(MavenPublication.class, publication -> {
-            // Add git origin info to generated POM files
-            publication.getPom().withXml(PublishPlugin::addScmInfo);
-
-            // have to defer this until archivesBaseName is set
-            project.afterEvaluate(p -> publication.setArtifactId(getArchivesBaseName(project)));
-
-            // publish sources and javadoc for Java projects.
-            if (project.getPluginManager().hasPlugin("opensearch.java")) {
-                publication.artifact(project.getTasks().getByName("sourcesJar"));
-                publication.artifact(project.getTasks().getByName("javadocJar"));
+            // To exclude java artifcats for maven zip type publications
+            String zipPublicationSearch = "zip";
+            if (!publication.getName().toLowerCase().contains(zipPublicationSearch.toLowerCase())) {
```

Comment:
Is this necessary? It looks like you're creating a second kind of publication (called `zippublish`) that is different from existing publication (called `publish`). Which is fine, but I would expect the publish plugin to not be touched/modified at all, or see Zips be published the same way as `sourcesJar` and `javadocJar` with a 1-liner of `publication.artifact(project.getTasks().getByName("....Zip"));`.

Reply:
I dont need to do zipPublicationSearch check anymore as I'm suspending the tasks in the ZipPublish.java. Please check my latest commit.